workflow/lib/preprocess/file_utils.py

```diff
@@ -25,60 +26,82 @@ def get_sample_fps(
+            if channel is not None:
+                filtered_df = filtered_df[filtered_df["channel"] == channel]
+        
         # Create dictionary mapping round to DataFrame rows
         round_groups = {
             round_num: group for round_num, group in filtered_df.groupby("round")
         }
-
-        # Initialize list to store file paths for each round
-        round_files = []
-
-        # Process each round in the specified order
-        for round_num in round_order:
-            if round_num not in round_groups:
-                raise ValueError(f"Round {round_num} not found in data")
-
+        
+        # Initialize lists to store files and channels
+        all_files = []
+        final_channel_order = []
+        
+        # Get available rounds that exist in the data
+        available_rounds = sorted(round_groups.keys())
+        
+        # Process each available round
+        for round_num in available_rounds:
             round_df = round_groups[round_num]
-
-            # If we have channels and channel order is specified
+            
+            # If channel order is specified, get files in that order for this round
             if "channel" in round_df.columns and channel_order is not None:
+                # Create mapping of available channels to files for this round
                 channel_to_file = dict(zip(round_df["channel"], round_df["sample_fp"]))
-                round_files.extend(
-                    [channel_to_file[channel] for channel in channel_order]
-                )
+                # Add files for each requested channel if available in this round
+                for channel in channel_order:
+                    if channel in channel_to_file:
+                        all_files.append(channel_to_file[channel])
+                        final_channel_order.append(f"Round {round_num}: {channel}")
             else:
-                round_files.append(round_df["sample_fp"].iloc[0])
-
-        return round_files
+                # If no channel order, just take the first file from this round
+                all_files.append(round_df["sample_fp"].iloc[0])
+                final_channel_order.append(f"Round {round_num}: {round_df['channel'].iloc[0]}")
+        
+        if verbose:
+            print("\nFinal channel order:")
+            for chan in final_channel_order:
+                print(f"  {chan}")
+                
+        return all_files
 
     # If no rounds specified but we have channels and channel order
     if "channel" in filtered_df.columns and channel_order is not None:
         channel_to_file = dict(zip(filtered_df["channel"], filtered_df["sample_fp"]))
-        return [channel_to_file[channel] for channel in channel_order]
+        return [channel_to_file[channel] for channel in channel_order if channel in channel_to_file]
 
     # Otherwise return single file path
-    return filtered_df["sample_fp"].iloc[0]
+    return filtered_df["sample_fp"].iloc[0]
```

Comment:
Missing empty line at EOF here. Did you make sure to run ruff check/format across brieflow? If so it might be some issue with your ruff config because these are caught for me.